### ISSUE TITLE
SC2004 false positive on $(( ${foo+bar} ))

### ISSUE THREAD

**Issue by xPMo** [koalaman/shellcheck#1533]
- SC2004
- Version 0.6.0 (shellcheck-static from AUR)
- Also occurs on shellcheck.net
- Related issue: #868 

#### Here's a snippet or screenshot that shows the problem:

```bash
#!/usr/bin/env sh
x=1

# ShellCheck incorrectly warns here:
echo $(( ${x+1} ))  # ⇒ 1

echo $(( x+1 ))     # ⇒ 2
```

Same error  for `bash`

#### Here's what shellcheck currently says:

```
In sc2004.sh line 5:
echo $(( ${x+1} ))  # ⇒ 1
         ^----^ SC2004: $/${} is unnecessary on arithmetic variables.
```

**Comment by koalaman** [comment]
Well caught, thanks! This has now been fixed, as has the similar cases for `${x=1}`, `${x,}` and `${x^}`
